sway-lsp/src/sway_config.rs

```diff
@@ -34,11 +42,24 @@ impl SwayConfig {
 // note `FormattingOptions` and `SwayConfig` may be similar at this moment,
 // but they are not the same thing, `SwayConfig` contains all the data related to the LanguageServer
 // while `FormattingOptions` is only the part that is necessary for 'formating'
-impl From<SwayConfig> for FormattingOptions {
+impl From<SwayConfig> for Formatter {
```

Comment:
Perhaps a closer mapping to the old behaviour would be:
```suggestion
impl From<SwayConfig> for Config {
```
where `Config` is the formatter's `Config` type?

Eventually we might want to simply remove `SwayConfig` in favour of using the formatter's `Config` directly, but we can discuss this with @JoshuaBatty once he's back in the office. For now, I think you're on the right track with simply using the defaults where necessary :+1:

Reply:
Checked in on this and while converting to `Config` may be closer to the previous code, it doesn't seem that we're using it anywhere else other than for the end result of constructing a `Formatter` and calling `.format()`. In other words we would be constructing a `Config` and then construct a `Formatter` and then replace the formatter's config with said config.